sdk/core/perf/inc/azure/perf/base_test.hpp

```diff
@@ -51,7 +51,14 @@ namespace Azure { namespace Perf {
      *
      * @param proxy A test-proxy server url.
      */
-    void SetTestProxy(std::string const& proxy) { m_proxy = proxy; }
+    void SetTestProxy(std::string const& proxy)
+    {
+      if (!proxy.empty())
+      {
+        m_isInsecureEnabled = true;
```

Comment:
Can you add a comment, why are you enabling Insecure.

I am reading the code, if the proxy is not empty, enable insecure (the name is also not very explanatory, in my opinion). Enable unsecured connections, I suppose. But still, why. How does that relate to proxy not being empty?

Is my understanding correct? - 
```
We want all the test that are using a test proxy
to communicate via unsecured HTTP, while when the test
is targeting the real Azure service endpoint, we want the secured HTTPS.
```

Is my understanding correct? I'd put a comment like this above this line.
I know all is clear when you write the code, but to me as a reader who is not very familiar with the code, it is not obvious at all what is going on. Assuming I even guessed it right :)